SVCB improvements (#10074)
PowerDNS/pdns: docs/guides/svcb.rst

Using SVCB and derived records
==============================

> The PowerDNS Authoritative Server has support for the SVCB and derived records (like HTTPS).

Reviewer: ```suggestion
The PowerDNS Authoritative Server has support for the SVCB record and derived records (like HTTPS).
```

This support includes doing the standards recommended following of alias-form records in-zone and adding those to the additional section.
Apart from that, there's the PowerDNS special for "autohints".

Automatic hints
---------------

> PowerDNS can automatically fill in ``ipv4hint`` and ``ipv6hint`` parameters in SVCB records based on A and AAAA records already in the zones.

Reviewer: ```suggestion
PowerDNS can automatically fill in ``ipv4hint`` and ``ipv6hint`` parameters in SVCB records based on A and AAAA records already present in the zone.
```

Consider the following zone content::

  example.com      IN HTTPS 0 www.example.com

  www.example.com  IN HTTPS 1 . ipv4hint=auto
  www.example.com  IN A     192.0.2.1
  www.example.com  IN AAAA  2001:db8::1

  ipv6.example.com IN HTTPS 1 . ipv6hint=auto
  ipv6.example.com IN AAAA  2001:db8::2
  ipv6.example.com IN AAAA  2001:db8::3

> A query for example.org|HTTPS will be responded to like this::

Reviewer: ```suggestion
A query for ``HTTPS example.org`` will be responded to like this::
```

  ;; QUESTION SECTION:
  ;example.org.			IN	HTTPS

  ;; ANSWER SECTION:
  example.org.		3600	IN	HTTPS	0 www.example.com

  ;; ADDITIONAL SECTION:
  www.example.org.	3600	IN	A	192.0.2.1
  www.example.org.	3600	IN	HTTPS	1 . ipv4hint=192.0.2.1
  www.example.org.	3600	IN	AAAA	2001:db8::1

> Notice that PowerDNS did additional processing and added the target HTTPS record,
> and the address records to the additional section.

Reviewer: ```suggestion
Notice that PowerDNS did additional processing and added the target HTTPS record, and the address records, to the additional section.
```

> Also notice that the ipv4hint in the www.example.org|HTTPS records is set to the value of the A record.

Reviewer: ```suggestion
Also notice that the ipv4hint in the ``www.example.org HTTPS`` records is set to the value of the A record.
```

Using SVCB and derived records
==============================
The PowerDNS Authoritative Server has support for the SVCB record and derived records (like HTTPS).
This support includes doing the standards recommended following of alias-form records in-zone and adding those to the additional section.
Apart from that, there's the PowerDNS special for "autohints".

.. _svc-autohints:

Automatic hints
---------------
PowerDNS can automatically fill in ``ipv4hint`` and ``ipv6hint`` parameters in SVCB records based on A and AAAA records already present in the zone.
This can be enabled by setting :ref:`setting-svc-autohint` to 'yes'.

Consider the following zone content::

  example.com      IN HTTPS 0 www.example.com

  www.example.com  IN HTTPS 1 . ipv4hint=auto
  www.example.com  IN A     192.0.2.1
  www.example.com  IN AAAA  2001:db8::1

  ipv6.example.com IN HTTPS 1 . ipv6hint=auto
  ipv6.example.com IN AAAA  2001:db8::2
  ipv6.example.com IN AAAA  2001:db8::3

A query for ``example.org|HTTPS`` will be responded to like this::

  ;; QUESTION SECTION:
  ;example.org.			IN	HTTPS

  ;; ANSWER SECTION:

>   example.org.		3600	IN	HTTPS	0 www.example.com

Reviewer: This says `.com` but the next few lines say `.org`

  ;; ADDITIONAL SECTION:
  www.example.org.	3600	IN	A	192.0.2.1
  www.example.org.	3600	IN	HTTPS	1 . ipv4hint=192.0.2.1
  www.example.org.	3600	IN	AAAA	2001:db8::1

Notice that PowerDNS did additional processing and added the target HTTPS record, and the address records to the additional section.
Also notice that the ipv4hint in the ``www.example.org|HTTPS`` records is set to the value of the A record.

PowerDNS takes all the corresponding address records, e.g. when asking for ipv6.example.com::

  ;; QUESTION SECTION:

>   ;ipv6.example.org.		IN	HTTPS

Reviewer: More org/com confusion here.
